### PR TITLE
feat: allow mixed mode Availability Zone configuration

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

This PR removes the "you must have Availability Zones configuration on masters + all node pools", and replaces that with warning guidance when such scenarios are detected.

This unblocks AZ scenarios for folks using other configurations in a subset of their cluster configuration (e.g., VM SKUs) for which Availability Zones is not possible, and allows those scenarios to get partial AZ support on the remaining relevant aspects of their cluster configuration.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by jackfrancis** [comment]
@ritazh FYI

File: examples/e2e-tests/kubernetes/release/default/definition.json [hunk 1]
```diff
@@ -50,7 +50,8 @@
       "OSDiskSizeGB": 200,
       "vnetSubnetId": "/subscriptions/SUB_ID/resourceGroups/RG_NAME/providers/Microsoft.Network/virtualNetworks/VNET_NAME/subnets/SUBNET_NAME",
       "firstConsecutiveStaticIP": "10.239.255.239",
-      "vnetCidr": "10.239.0.0/16"
+      "vnetCidr": "10.239.0.0/16",
+      "availabilityZones": ["1", "2"]
```
Comment (jackfrancis) [inline examples/e2e-tests/kubernetes/release/default/definition.json:54]: This change means we are validating this change in this PR (i.e., we are setting the cluster config that PRs run E2E against for Availability Zones 1 and 2 on master VMs only, but not on any node pools. Previously this was not allowed.

File: pkg/api/vlabs/validate_test.go [hunk 2]
```diff
@@ -2998,18 +3006,121 @@ func TestProperties_ValidateZones(t *testing.T) {
...
+		Count:               5,
+		DNSPrefix:           "foo",
+		VMSize:              "Standard_DS2_v2",
+		AvailabilityProfile: VirtualMachineScaleSets,
+		AvailabilityZones:   []string{"1", "2"},
+	}
+	cs.Properties.validateZones()
+	// Should yield:
+	// level=warning msg="This cluster is using Availability Zones for master VMs, but not for pools \"anotherpool\" and \"anotherpool3\""
+	// The ordered collection of all output is validated below:
```
Comment (mboersma) [inline pkg/api/vlabs/validate_test.go:3115]: Nice use of an Example_ test here.

File: pkg/helpers/helpers.go [hunk 3]
```diff
@@ -258,3 +258,28 @@ func GetLogAnalyticsWorkspaceDomain(cloudOrDependenciesLocation string) string {
 	}
 	return workspaceDomain
 }
+
+// GetEnglishOrderedQuotedListWithOxfordCommas returns a string that you can use in an English statement to report a list of things
+// With each thing in quotes, e.g. "thing 1", "thing 2", and "thing 3"
+// Oxford Comma not optional!
```
Comment (mboersma) [inline pkg/helpers/helpers.go:264]: Agreed.

**Pull request merged**